src/App.js

```diff
@@ -15,7 +16,20 @@ function App() {
     <div className="App">
       <Routes>
         <Route path="/" element={<Layout />}>
-          <Route index element={user ? <Home /> : <Navigate to="/signin" />} />
+          <Route
+            index
+            element={
+              user ? <Navigate to="/notes" /> : <Navigate to="/signin" />
+            }
+          />
+          <Route
+            path="notes"
+            element={user ? <Home /> : <Navigate to="/signin" />}
+          />
+          <Route
+            path="notes/:noteId"
+            element={user ? <SingleNote /> : <Navigate to="/signin" />}
+          />
```

Comment:
You are performing here the same check over and over again, please find a way to generalize this behavior. For example, you can create a custom routing component called ProtectedRoute which internally checks if the use is logged in.